common/compositor/va/varenderer.cpp

```diff
@@ -216,14 +216,17 @@ bool VARenderer::SetVAProcFilterDeinterlaceMode(const HWCDeinterlaceProp& prop,
 }
 
 unsigned int VARenderer::GetVAProcFilterScalingMode(uint32_t mode) {
-  switch (mode) {
-    case 1:
-      return VA_FILTER_SCALING_FAST;
-    case 2:
-      return VA_FILTER_SCALING_HQ;
-    default:
-      return VA_FILTER_SCALING_HQ;
-  }
+  if (deinterlace_caps_.mode_ == VAProcDeinterlacingNone) {
```

Comment:
I'm not sure why you're not returning the filter scaling mode if deinterlace_caps_.mode_== VAProcDeinterlacingBob or VAProcDeinterlacingWeave.  I think you're trying to force the mode to FAST, but first time around it will not be FAST in case (mode==2) && (deinterlace_caps_.mode_ == VAProcDeinterlacingNone)

Reply:
I am not getting your point

Reply:
All cases are returned, we have if else